### PR TITLE
add logger to mlr_reflections

### PR DESCRIPTION
**Pull request by be-marc**
https://github.com/mlr-org/mlr3/pull/736

### CODE REVIEW

**Comment by mllg** [comment]
This won't work if you first load bbotk and then mlr3. You need an onload hook.

**Comment by mllg** [comment]
I think you can use https://github.com/mlr-org/mlr3misc/blob/main/R/register_namespace_callback.R.

**Comment by be-marc** [comment]
Forget to mention that but okay there is a magic function for everything in `mlr3misc`. Thanks!

File: R/zzz.R [hunk 1]
```diff
@@ -17,6 +17,11 @@
   }
   lg$set_filters(f)
 
+  register_namespace_callback("mlr3", "mlr3", function(pkgname, pkgpath) {
```
Comment (mllg) [inline R/zzz.R:20]: First argument is the package which is registering the callback, i.e. "bbotk".

Comment (mllg) [reply R/zzz.R:20]: (or `pkgname`)

**Pull request merged**